app/models/user.rb

```diff
@@ -0,0 +1,27 @@
+class User < ApplicationRecord
+  has_secure_password
+  # Validations
+  validates :name, presence: true, length: { maximum: 30 }
+  validates :email, presence: true, length: { maximum: 50 }, uniqueness: true, format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
+
+  generates_token_for :password_reset, expires_in: 15.minutes do
+    password_salt&.last(10)
+  end
+
+  generates_token_for :email_confirmation, expires_in: 24.hours do
+    email
+  end
+
+  def confirm!
+    update_columns(confirmed_at: Time.current)
+  end
+
+  def confirmed?
+    confirmed_at.present?
+  end
+
+
```

Comment:
additional line space? you can try [rubocop](https://github.com/rubocop/rubocop).